implementations/rust/ockam/ockam_api/src/cli_state/storage/projects_repository_sql.rs

```diff
@@ -264,6 +303,17 @@ impl ProjectsRepository for ProjectsSqlxDatabase {
 
     async fn set_default_project(&self, project_id: &str) -> Result<()> {
         let mut transaction = self.database.begin().await.into_core()?;
+
+        if let Some(project) = self.get_project(project_id).await? {
+            if self.is_shared_project(&project, &mut transaction).await? {
+                warn!(
+                    "the project {} can't be set as default because is not owned by any local user",
+                    project_id
+                );
+                return Ok(());
```

Comment:
I think we should return an error in that case. The user wants to set the project as a default but the action was not performed, this should not stay silent.

Reply:
The non-admin projects are hidden from the user when using the command, so they shouldn't be able to interact directly with this kind of projects.

The reason I did this is that we have [some logic](https://github.com/build-trust/ockam/blob/b1f9800944f9f1cc3e7915052eb13ee4dc95c736/implementations/rust/ockam/ockam_api/src/cli_state/projects.rs#L74) where we set a new default project after deleting another project. Would you rather return an error in the repository and ignore the error in this layer?

Reply:
> The non-admin projects are hidden from the user when using the command, so they shouldn't be able to interact directly with this kind of projects.

I see.

> The reason I did this is that we have [some logic](https://github.com/build-trust/ockam/blob/b1f9800944f9f1cc3e7915052eb13ee4dc95c736/implementations/rust/ockam/ockam_api/src/cli_state/projects.rs#L74) where we set a new default project after deleting another project. Would you rather return an error in the repository and ignore the error in this layer?

Oh, but then the logic which tries to set a default project could select a shared project as the first one, try to set it as a default, then silently fail and we would end-up with no default project?
I think this means that the `delete_project` should take care of this logic inside `projects_repository_sql`. And this will be better because it will be inside the same transaction.

Reply:
Done!